src/main/scala/org/broadinstitute/hail/expr/FunctionRegistry.scala

```diff
@@ -105,110 +110,113 @@ object FunctionRegistry {
+      //   val localA = ec.a
+      //   val idx = localA.length
+      //   localA += null
+
+      //   val u = args(0).eval(EvalContext())()
+
+      //   if (u == null)
+      //     fatal("Argument evaluated to missing in call to aggregator $name")
+
+      //   ec.aggregations += ((idx, lhs.evalAggregator(ec), aggregator.ctor(
+      //     u.asInstanceOf[u])))
+      //   () => localA(idx)
+
+      // case aggregator: Arity3Aggregator[_, u, v, w, _] =>
+      //   val localA = ec.a
+      //   val idx = localA.length
+      //   localA += null
+
+      //   val u = args(0).eval(EvalContext())()
+      //   val v = args(1).eval(EvalContext())()
+      //   val w = args(2).eval(EvalContext())()
+
+      //   if (u == null)
+      //     fatal("Argument 1 evaluated to missing in call to aggregator $name")
+      //   if (v == null)
+      //     fatal("Argument 2 evaluated to missing in call to aggregator $name")
+      //   if (w == null)
+      //     fatal("Argument 3 evaluated to missing in call to aggregator $name")
+
+      //   ec.aggregations += ((idx, lhs.evalAggregator(ec), aggregator.ctor(
+      //     u.asInstanceOf[u],
+      //     v.asInstanceOf[v],
+      //     w.asInstanceOf[w])))
+      //   () => localA(idx)
+
+      // case aggregator: UnaryLambdaAggregator[t, u, v] =>
+      //   val Lambda(_, param, body) = args(0)
+
+      //   val idx = ec.a.length
+      //   val localA = ec.a
+      //   localA += null
+
+      //   val bodyST =
+      //     lhs.`type` match {
+      //       case tagg: TAggregable => tagg.symTab
+      //       case _ => ec.st
+      //     }
+
+      //   val bodyFn = body.eval(ec.copy(st = bodyST + (param -> (idx, lhs.`type`.asInstanceOf[TContainer].elementType))))
+      //   val g = (x: Any) => {
+      //     localA(idx) = x
+      //     bodyFn()
+      //   }
+
+      //   ec.aggregations += ((idx, lhs.evalAggregator(ec), aggregator.ctor(g)))
+      //   () => localA(idx)
+
+      // case aggregator: BinaryLambdaAggregator[t, u, v, w] =>
+      //   val Lambda(_, param, body) = args(0)
+
+      //   val idx = ec.a.length
+      //   val localA = ec.a
+      //   localA += null
+
+      //   val bodyST =
+      //     lhs.`type` match {
+      //       case tagg: TAggregable => tagg.symTab
+      //       case _ => ec.st
+      //     }
+
+      //   val bodyFn = body.eval(ec.copy(st = bodyST + (param -> (idx, lhs.`type`.asInstanceOf[TContainer].elementType))))
+      //   val g = (x: Any) => {
+      //     localA(idx) = x
+      //     bodyFn()
+      //   }
+
+      //   val v = args(1).eval(EvalContext())()
+      //   if (v == null)
+      //     fatal("Argument evaluated to missing in call to aggregator $name")
+
+      //   ec.aggregations += ((idx, lhs.evalAggregator(ec), aggregator.ctor(g, v.asInstanceOf[v])))
+      //   () => localA(idx)
 
       case f: UnaryFun[_, _] =>
         AST.evalCompose(ec, lhs)(f)
+      case f: UnaryFunCode[_, _] => ???
```

Comment:
Check out `AST.scala`. I just recreated the function registry there but for the compiler. Eventually I should move it back here and delete all this old stuff.